include/ibamr/AllenCahnHierarchyIntegrator.h

```diff
@@ -0,0 +1,337 @@
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_Div_u_var;
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_T_lf_N_var;
+
+    /*!
+     * Cartgrid functions to be used to set the Allen-Cahn equation source term.
+     */
+    SAMRAI::tbox::Pointer<IBTK::CartGridFunction> d_lf_F_fcn;
+    SAMRAI::tbox::Pointer<IBTK::CartGridFunction> d_lf_F_init;
+
+    /*!
+     * Boundary condition object for \f$ \varphi \f$.
+     */
+    SAMRAI::solv::RobinBcCoefStrategy<NDIM>* d_lf_bc_coef = nullptr;
+
+    /*!
+     * Convective operator and difference type for the Allen-Cahn equation.
+     */
+    ConvectiveDifferencingType d_lf_convective_difference_form = d_default_convective_difference_form;
+    std::string d_lf_convective_op_type = d_default_convective_op_type;
+    SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> d_lf_convective_op_input_db = d_default_convective_op_input_db;
+    SAMRAI::tbox::Pointer<ConvectiveOperator> d_lf_convective_op = nullptr;
+    bool d_lf_convective_op_needs_init = false;
+
+    /*!
+     * Hierarchy operators, solvers and related data for the Allen-Cahn equation.
+     */
+    std::string d_lf_solver_type = IBTK::CCPoissonSolverManager::UNDEFINED,
+                d_lf_precond_type = IBTK::CCPoissonSolverManager::UNDEFINED;
+    SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> d_lf_solver_db, d_lf_precond_db;
+    SAMRAI::tbox::Pointer<IBTK::LaplaceOperator> d_lf_rhs_op = nullptr;
+    SAMRAI::tbox::Pointer<IBTK::PoissonSolver> d_lf_solver = nullptr;
+    bool d_lf_solver_needs_init, d_lf_rhs_op_needs_init;
+    SAMRAI::tbox::Pointer<SAMRAI::solv::SAMRAIVectorReal<NDIM, double> > d_lf_sol, d_lf_rhs;
+
+    TimeSteppingType d_lf_diffusion_time_stepping_type = d_default_diffusion_time_stepping_type;
+    TimeSteppingType d_lf_convective_time_stepping_type = d_default_convective_time_stepping_type;
+    TimeSteppingType d_lf_init_convective_time_stepping_type = d_default_convective_time_stepping_type;
+
+    /*!
+     * Boolean to identify whether the energy equation is to be solved.
+     */
+    bool d_solve_energy = false;
+
+    /*!
+     * Patch data descriptor indices for all "scratch" variables managed by the
+     * integrator.
+     *
+     * Scratch variables have only one context: scratch.
+     */
+    int d_lf_temp_rhs_idx = IBTK::invalid_index, d_lf_C_idx = IBTK::invalid_index;
+    int d_g_firstder_idx = IBTK::invalid_index, d_g_secondder_idx = IBTK::invalid_index,
+        d_q_firstder_idx = IBTK::invalid_index;
+    int d_chemical_potential_idx = IBTK::invalid_index, d_grad_lf_idx = IBTK::invalid_index;
+    int d_H_sc_idx = IBTK::invalid_index;
+    int d_T_lf_N_scratch_idx = IBTK::invalid_index;
+
+    /*!
+     * Allen-Cahn equation parameters.
+     * M_lf  - Mobility
+     * lambda_lf - Mixing energy density
+     * eps_lf - Thickness of the liquid-solid interface
+     */
+    double d_M_lf, d_lambda_lf, d_eps_lf;
```

Comment:
These should have default values, even if it's just `std::numeric_limits<double>::signaling_NaN()`.